.github/workflows/pr.yml

```diff
@@ -68,35 +50,17 @@ jobs:
       fail-fast: false
       matrix:
         java: [ 11, 17 ]
+    if: contains(github.ref, 'gh-')  # For now, only run tests on "gh-*" PRs
     name: JUnit tests Java ${{ matrix.java }}
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-        with:
-          persist-credentials: false
-      - name: Setup Java
-        uses: actions/setup-java@v4
-        with:
-          distribution: temurin
-          java-version: ${{ matrix.java }}
       - name: Setup Gradle
-        uses: gradle/actions/setup-gradle@v4
-        env:
-          GRADLE_BUILD_ACTION_CACHE_DEBUG_ENABLED: true
+        uses: ./.github/actions/setup-gradle
         with:
-          gradle-version: wrapper
-          develocity-access-key: ${{ secrets.GE_ACCESS_TOKEN }}
-          # Only write to the cache from trunk
-          #cache-read-only: ${{ github.ref != 'refs/heads/trunk' }}
-          cache-read-only: false
-          # Cache downloaded JDKs in addition to the default directories.
-          gradle-home-cache-includes: |
-            caches
-            notifications
-            jdks
-          cache-cleanup: on-success
-          cache-overwrite-existing: true
+          java-version: ${{ matrix.java }}
+          gradle-cache-read-only: true
       - name: Test
+        # Publish build scans for now. This will only work on PRs from apache/kafka, not public forks.
```

Comment:
Do we open PRs from apache/kafka? it seems almost all PRs are from public forks.

Also, the failed tests in PR phase are expected. Maybe publishing the scans in CI phase is more acceptable since that is what running in code base.

Reply:
I think all our PRs come from forks normally. Leaving this in here does give committers a way to get a build scan from a non-trunk code base.